Indicators/SharpeRatio.cs

```diff
@@ -0,0 +1,129 @@
+        /// </summary>
+        public int SharpePeriod { get; }
+
+        /// <summary>
+        /// Stores the value of the risk-free rate of return for calculation
+        /// </summary>
+        public decimal RiskFreeRate { get; }
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Returns whether the indicator is properly initalized with data
+        /// </summary>
+        public override bool IsReady => _values.IsReady && _std.IsReady;
+
+        /// <summary>
+        /// Creates a new SharpeRatio indicator using the specified periods
+        /// </summary>
+		/// <param name="name">The name of this indicator</param>
+		/// <param name="sharpePeriod">Period of historical observation for sharpe ratio calculation</param>
+		/// <param name="riskFreeRate">Risk-free rate for sharpe ratio calculation</param>
+        public SharpeRatio(string name, int sharpePeriod, decimal riskFreeRate = 0.0m)
+            : base(name)
+        {
+            // init private variables
+            _std = new StandardDeviation(sharpePeriod);
+            _values = new RollingWindow<IndicatorDataPoint>(sharpePeriod + 1);
+
+            // init public variables
+            SharpePeriod = sharpePeriod;
+            RiskFreeRate = riskFreeRate;
+            WarmUpPeriod = sharpePeriod + 1;
+        }
+
+        /// <summary>
+        /// Creates a new SharpeRatio indicator using the specified periods
+        /// </summary>
+        /// <param name="sharpePeriod">Period of historical observation for sharpe ratio calculation</param>
+		/// <param name="riskFreeRate">Risk-free rate for sharpe ratio calculation</param>
+        public SharpeRatio(int sharpePeriod, decimal riskFreeRate = 0.0m)
+            : this($"SR({sharpePeriod})", sharpePeriod, riskFreeRate)
+        {
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// </summary>
+		/// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IndicatorDataPoint input)
+        {
+            // update indicators
+            _values.Add(input);
+            _std.Update(input);
+
+            // check for IsReady
+            if (!IsReady)
+                return 0.0m;
+
+            // calculates SharpeRatio
+            // makes sure no divisibilty errors occur
+            // define percent change from prior value SharpePeriod values away
+            decimal percentChange = input.Value != 0.0m ? ((input.Value - _values[SharpePeriod].Value) / input.Value) : 0.0m;
+            // define std as a percent value over input
+            decimal stdAsPercentage = input.Value != 0.0m ? _std / input.Value : 0.0m;
```

Comment:
I think that the average of the rate of change over a given period is the same as the rate of change over the period, so the calculation of `percentageChange` is right. However, the same might no be true for `stdAsPercentage`. Did you compare the results from this against `QuantConnect.Statistics.Statistics.ObservedSharpeRatio`?

Reply:
Thank you for the review Alex. While checking, I realized that the TradingView script I modeled the calculation on was slightly wrong. I'll fix the issues and submit an update.